internal/pkg/runtime/engine/singularity/container_linux.go

```diff
@@ -2139,9 +2139,7 @@ func (c *container) addFilesMount(system *mount.System) error {
 }
 
 func (c *container) addIdentityMount(system *mount.System) error {
-	if (os.Geteuid() == 0 && c.engine.EngineConfig.GetTargetUID() == 0) ||
-		c.engine.EngineConfig.GetFakeroot() {
-		sylog.Verbosef("Not updating passwd/group files, running as root!")
+	if c.engine.EngineConfig.GetNoHome() {
```

Comment:
Sorry - on a second pass over it I don't think that this is what we want.

`--no-home` is just about not mounting the HOME directory into the container.

It's actually (confusingly) valid to do something like `--no-home --home /opt`... in which case HOME should be `/opt` in the container... but not  `/opt` will not be mounted from outside.

Also, if I'm user `dave`, then even if I run with `--no-home` then we need to make sure `dave` is present in the container `/etc/passwd`.

Apologies for not picking this up before. I will need to find a couple of hours to think about this and trawl through some old discussions from a few years ago to satisfy myself what the right approach is.

Reply:
Right... having combed through the code and the history of this in the morning... it looks to me like we should just always update the passwd/group file.

Let's do that... we're swapping 'don't update only if root' for 'always update'. On a Linux host you can certainly change root's home directory from `/root` if you want to. It's not necessarily a sensible thing, but you can do it... so `--home` should do it too, for consistency.

If I've missed anything... we've got a few months for it to surface before this goes out in a 4.0 :-)